crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1010,11 +1021,24 @@ impl Analyzer<'_, '_> {
                         ..
                     }) => {
                         // validEnumAssignments.ts insists that this is valid.
-                        return Ok(());
+                        // but if enum isn't has num, not assignable
+                        let items = self.find_type(enum_name).context("failed to find an enum for assignment")?;
+
+                        if let Some(items) = items {
+                            for t in items {
+                                if let Type::Enum(en) = t.normalize() {
+                                    if en.has_num {
+                                        return Ok(());
+                                    }
+                                }
+                            }
+                        }
+
+                        fail!()
                     }
 
                     Type::EnumVariant(rhs) => {
-                        if rhs.enum_name == enum_name {
+                        if rhs.enum_name == enum_name.clone() {
```

Comment:
Please change this to `rhs.enum_name == *enum_name`

Reply:
I guess I was hasty. I agree.
Thank you for your feedback.